packages/core/src/ceramic.ts

```diff
@@ -178,15 +178,15 @@ class Ceramic implements CeramicApi {
   }
 
   private static async _generateNetworkOptions(config: CeramicConfig, anchorService: AnchorService): Promise<CeramicNetworkOptions> {
-    const networkName = config.networkName || DEFAULT_NETWORK
```

Comment:
Why mutation like this? It is effect could be negligible here, but it certainly introduces new risk down the evolution line.

Aha, if I understand correctly, this modifies `config` here to only pass proper network name to `PinStoreFactory`.

May I suggest instead of this to explicitly create a new `effectiveConfig` (or something like it) variable that sets all the defaults for the internal consumers like PinStoreFactory and Ceramic class itself? Being explicit here makes it clear what configuration is effectively used, where it is set and based on what.

For pin-store it makes sense to extend internal/effective config with `pinStorePath` property that is set to appropriate stateStorePath + suffix + network.

Data flow becomes more clear cut `configuration set by caller -> effective configuration -> consumers`.

Also, mea culpa, `process.cwd` should have never been used inside PinStoreFactory. It should silently accept whatever path it is constructed with.